framework/src/multiapps/FullSolveMultiApp.C

```diff
@@ -23,12 +23,17 @@ FullSolveMultiApp::validParams()
 {
   InputParameters params = MultiApp::validParams();
   params.addClassDescription("Performs a complete simulation during each execution.");
-  params.addParam<bool>(
+  params.addDeprecatedParam<bool>(
       "no_backup_and_restore",
       false,
-      "True to turn off backup/restore for this multiapp. This is useful when doing steady-state "
+      "True to turn off restore for this multiapp; use 'no_restore' instead.",
+      "This functionality has been replaced with 'no_restore' and now only disables restoration.");
+  params.addParam<bool>(
```

Comment:
use params.addParam params.deprecateParam instead (see ParsedAux for an example)

then you can remove the first check in the constructor